### DISCUSSION TITLE
Include related classes when processing specific files

### DISCUSSION THREAD

**Discussion by juanariastn** [rectorphp/rector#8766] [Q&A] [answered]
Hi all! Thanks for the tool, it works great! 🙏 

# Context
I have a massive project with thousands of files moving from PHP 7.0 to 7.1 and I want to execute an step in my CI to detect incompatible changes before we merge them.

I configured my CI step to check only for modified files using:
```
CHANGED_FILES=(use git commands to detect modified files in current branch)
vendor/bin/rector process $CHANGED_FILES
```

# Issue
Let's say I have these files:
- `Client.php`
- `Response.php`

And my scenario:
- I change the file `Client.php` adding a call to a method defined in `Response.php` with more parameters than the method defines
- That should break the rule `RemoveExtraParametersRector`. And indeed it does ;) 

## Working scenario
If I pass both files to process, `rector` detects the issue (which indicates the problem is not related to autolading of classes, right?):
```
vendor/bin/rector process Client.php Response.php
```

## Non working scenario
But since I've only modified the class `Client.php`, then the rector command is only executed with:
```
vendor/bin/rector process Client.php
```

That command doesn’t detect any issue.

Is there a way to indicate `rector` to include those extra files referenced by the file I specified to analyse?
Am I missing something? Is this the expected behavior?

Thanks in advance!!

**Comment by samsonasik** [comment, marked as answer]
Hi,

That's not possible, rector only work on target analyzed file, if you want multi files, then both files need to be included, on project basis, you can include by dirs collection:

```php
withPaths([
     __DIR__ . '/src',
     __DIR__ . '/test'
])
```

For some circumstance, you may need to use `--clear-cache` :)
